Sources/SwiftDocC/Model/Identifier.swift

```diff
@@ -92,30 +96,34 @@ public struct ResolvedTopicReference: Hashable, Codable, Equatable, CustomString
         return url?.scheme?.lowercased() == ResolvedTopicReference.urlScheme
     }
     
+    /// The storage for the resolved topic reference's state.
+    let _storage: Storage
+    
     /// The identifier of the bundle that owns this documentation topic.
     public var bundleIdentifier: String {
-        didSet { updateURL() }
+        return _storage.bundleIdentifier
```

Comment:
What performance gain do we get by making this type immutable? Thinking about this more, I prefer the flexibility of keeping this type mutable. It allows for clients to easily mutate properties of this type without calling the initializer and configure all the properties again. If making it immutable provides a significant performance gain, I think we should look at alternatives to improve performance in the future. What do you think?

Also, as it stands, this type is not copy-on-write, because it cannot be written to.

Reply:
> Thinking about this more, I prefer the flexibility of keeping this type mutable. It allows for clients to easily mutate properties of this type without calling the initializer and configure all the properties again.

I think this is a good point and in general I agree. However, because `ResolvedTopicReference` is effectively a wrapper around URL I think it's important to expose a similar API to URL and, thus, communicate with clients that doing something like adding a path component is an expensive operation.

Prior to this, we exposed the `pathComponents` property directly and then monitored it to recalculate the URL when it was modified. This is likely surprising to clients who would not expect the direct modification of an array of strings to be an expensive operation. Limiting clients to using the `addingPathComponent(_:)` style api communicates better what's actually happening here.

I've added further documentation to `ResolvedTopicReference` in a commit here (5c78d85882c8a9e6c7770a35dcea174e23dd7b7e) to clarify this.

Reply:
That's a great argument. The fact that it's a wrapper around URL, it makes sense to convey similar characteristics to URL, which includes immutability. The alternative here would've been to keep it mutable and document the properties that incur a performance cost when written to, but I do prefer the immutable approach now. Thanks for adding the doc comment!